source/extensions/stat_sinks/common/hystrix/hystrix.cc

```diff
@@ -0,0 +1,320 @@
+                 max_concurrent_requests, cluster_info);
+  addIntToStream("propertyValue_fallbackIsolationSemaphoreMaxConcurrentRequests", 0, cluster_info);
+  addInfoToStream("propertyValue_requestCacheEnabled", "false", cluster_info);
+  addInfoToStream("propertyValue_requestLogEnabled", "true", cluster_info);
+  addIntToStream("reportingHosts", reporting_hosts, cluster_info);
+  addIntToStream("propertyValue_metricsRollingStatisticalWindowInMilliseconds", rolling_window,
+                 cluster_info);
+
+  ss << "data: {" << cluster_info.str() << "}" << std::endl << std::endl;
+}
+
+void Hystrix::addHystrixThreadPool(std::stringstream& ss, absl::string_view cluster_name,
+                                   uint64_t queue_size, uint64_t reporting_hosts,
+                                   uint64_t rolling_window) {
+  std::stringstream cluster_info;
+
+  addIntToStream("currentPoolSize", 0, cluster_info);
+  addIntToStream("rollingMaxActiveThreads", 0, cluster_info);
+  addIntToStream("currentActiveCount", 0, cluster_info);
+  addIntToStream("currentCompletedTaskCount", 0, cluster_info);
+  addIntToStream("propertyValue_queueSizeRejectionThreshold", queue_size, cluster_info);
+  addStringToStream("type", "HystrixThreadPool", cluster_info);
+  addIntToStream("reportingHosts", reporting_hosts, cluster_info);
+  addIntToStream("propertyValue_metricsRollingStatisticalWindowInMilliseconds", rolling_window,
+                 cluster_info);
+  addStringToStream("name", cluster_name, cluster_info);
+  addIntToStream("currentLargestPoolSize", 0, cluster_info);
+  addIntToStream("currentCorePoolSize", 0, cluster_info);
+  addIntToStream("currentQueueSize", 0, cluster_info);
+  addIntToStream("currentTaskCount", 0, cluster_info);
+  addIntToStream("rollingCountThreadsExecuted", 0, cluster_info);
+  addIntToStream("currentMaximumPoolSize", 0, cluster_info);
+
+  ss << "data: {" << cluster_info.str() << "}" << std::endl << std::endl;
+}
+
+void Hystrix::getClusterStats(std::stringstream& ss, absl::string_view cluster_name,
+                              uint64_t max_concurrent_requests, uint64_t reporting_hosts,
+                              uint64_t rolling_window) {
+  addHystrixCommand(ss, cluster_name, max_concurrent_requests, reporting_hosts, rolling_window);
+  addHystrixThreadPool(ss, cluster_name, max_concurrent_requests, reporting_hosts, rolling_window);
+}
+
+absl::string_view Hystrix::printRollingWindow() const {
+  std::stringstream out_str;
+
+  for (auto stats_map_itr = rolling_stats_map_.begin(); stats_map_itr != rolling_stats_map_.end();
+       ++stats_map_itr) {
+    out_str << stats_map_itr->first << " | ";
+    RollingStats rolling_stats = stats_map_itr->second;
+    for (auto specific_stat_vec_itr = rolling_stats.begin();
+         specific_stat_vec_itr != rolling_stats.end(); ++specific_stat_vec_itr) {
+      out_str << *specific_stat_vec_itr << " | ";
+    }
+    out_str << std::endl;
+  }
+  return out_str.str();
```

Comment:
@jmarantz can check me on this, but this seems like a dangerous/incorrect use of `absl::string_view`.   According to [the source](https://github.com/abseil/abseil-cpp/blob/master/absl/strings/string_view.h#L80), the `std::string` must outlive the `absl::string_view`.